server/data/data-api/src/main/java/org/apache/james/rrt/lib/Mapping.java

```diff
@@ -43,17 +43,21 @@ static Type detectType(String input) {
         if (input.startsWith(Type.Forward.asPrefix())) {
             return Type.Forward;
         }
+        if (input.startsWith(Type.Group.asPrefix())) {
+            return Type.Group;
+        }
         return Type.Address;
     }
 
     Optional<MailAddress> asMailAddress();
 
     enum Type {
-        Regex("regex:", 3),
```

Comment:
These orders don't make sense anymore.
My take is that we try to encode a policy into a sorting algorithm and having an int is convenient for sorting things.
Unfortunately the intent is very hard to guess.
Could we just have a Policy class ?

Reply:
done